Sources/AuthFoundation/Network/APIClient.swift

```diff
@@ -134,12 +220,21 @@ extension APIClient {
             }
         }.resume()
     }
+    
+    private func addRetryHeadersToRequest(state: APIRetry.State) -> URLRequest {
+        var request = state.originalRequest
+        if let requestId = state.requestId {
+            request.allHTTPHeaderFields?.updateValue(requestId, forKey: "X-Okta-Retry-For")
```

Comment:
Since the header is optional, we won't be throwing/catching any exception on the client side I don't think checking the X-Okta-Retry-For would add any benefit. I did add test for the `X-Okta-Retry-Count` header because the count is required.

Reply:
I think it's still worthwhile to have a test just for checking that we're adding the header correctly if it's present. It doesn't have to be in every test, like with X-Okta-Retry-Count

Reply:
I've updated this. Thanks